src/ProjectOrigin.Registry.IntegrationTests/ConsistentHashResolverTests.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.Extensions.Options;
+using ProjectOrigin.Registry.IntegrationTests.Extensions;
+using ProjectOrigin.Registry.Server.Options;
+using ProjectOrigin.Registry.Server.Services;
+using Xunit;
+
+namespace ProjectOrigin.Registry.IntegrationTests;
+
+public class ConsistentHashResolverTests
+{
+    [Fact]
+    public void GetQueueInRing_VerifyDistribution()
+    {
+        Random r = new Random(42); // Seed the random number generator to get consistent results
+        var options = new TransactionProcessorOptions
+        {
+            ServerNumber = 0,
+            Servers = 3,
+            Threads = 5,
+            Weight = 20
+        };
+        var number = 100000;
+        var expected = number / (options.Servers * options.Threads);
+        var resolver = new ConsistentHashRingQueueResolver(Options.Create(options));
+
+        ConcurrentDictionary<string, int> queueCounts = new ConcurrentDictionary<string, int>();
+        for (int i = 0; i < number; i++)
+        {
+            var s = r.GenerateString();
+            var queue = resolver.GetQueueName(s);
+            queueCounts.AddOrUpdate(queue, 1, (key, value) => value + 1);
+        }
+
+        double deviation = NormalizedStandardDeviation(queueCounts.Select(x => x.Value));
+
+        deviation.Should().BeLessThan(0.2);
+    }
+
+    [Fact]
+    public void GetQueue_ReturnsConsistentQueue()
+    {
+        var options = new TransactionProcessorOptions
+        {
+            ServerNumber = 0,
+            Servers = 2,
+            Threads = 2,
+            Weight = 2
+        };
+        var resolver = new ConsistentHashRingQueueResolver(Options.Create(options));
+
+        var queue = resolver.GetQueueName("test");
+
+        Assert.Equal("registry_0.verifier_1", queue);
+    }
```

Comment:
I see that you are not using the same amount of threads, weight, and replicas for tests as in prod. is this integration test then valid enough?

Reply:
this is a unit-test of the ConsistentHashRingQueueResolver and its functionality is tested in isolation.
(yes i know it is in a project named IntegrationTests, currently all tests are in the same project.)